### PR TITLE
Update Document/0x04a-Android.md - Added Permissions

### PR DESCRIPTION
**Pull request by luander**
Please review. Opinions and suggestions are welcome.

### CODE REVIEW

**Comment by clviper** [comment]
Hi @luander.

Thank you for the push. Please fix the typo in <uses-permisison /> to <uses-permissison />

File: Document/0x04a-Android.md [hunk 1]
```diff
@@ -160,6 +160,66 @@ A reference to the Messenger can be sent via an Intent using the previously ment
...
 
+#### Permissions
+Because Android applications are installed in a sandbox and initially it does not have access to neither user information nor access to system components (such as using the camera or the microphone), it provides a system based on permissions where the system has a predefined set of permissions for certain tasks that the application can request.
+As an example, if you want your application to use the camera on the phone you have to request the camera permission. 
+On Android versions before Marshmallow (API 23) all permissions requested by an application were granted at installation time. From Android Marshmallow onwards the user have to approve some permissions during application execution.
+
+##### Protection Levels
+Android permissions are classified in four different categories based on the protection level it offers.
+- *Normal*: Is the lower level of protection, it gives applications access to isolated application-level feature, with minimal risk to other applications, the user or the system. It is granted at installation. It is the default value.
```
Comment (sushi2k) [inline Document/0x04a-Android.md]: Maybe you say "They are granted during installation of the App."

What do you mean with "It is the default value."?

File: Document/0x04a-Android.md [hunk 2]
```diff
@@ -160,6 +160,66 @@ A reference to the Messenger can be sent via an Intent using the previously ment
...
+Example: `android.permission.INTERNET`
+- *Dangerous*: This permission usually gives the application control over user data or control over the device that impacts the user. This type of permissoin may not be granted at installation time, leaving to the user decide whether the application should have the permission or not.
+Example: `android.permission.RECORD_AUDIO`
+- *Signature*: This permission is granted only the the requesting app was signed with the same certificate as the application that declared the permission. If the signature matches, the permission is automatically granted.
```
Comment (sushi2k) [inline Document/0x04a-Android.md]: Two times "the". Maybe you write:
This permission is granted only to the requesting app if signed with the same certificate as the application that declared the permission.

File: Document/0x04a-Android.md [hunk 3]
```diff
@@ -160,6 +160,66 @@ A reference to the Messenger can be sent via an Intent using the previously ment
...
+    <permission
+    android:name="com.permissions.sample.ACCESS_USER_INFO"
+    android:protectionLevel="signature" />
+    <application>...</application>
+</manifest>
+```
+Only applications signed with the same developer certificate can use this permission.
+
+##### Enforcing Permissions on Android Components
+It is possible to protect Android components using permissions. Activities, Services, Content Providers and Content Providers all can use the permission mechanism to protect its interfaces.
```
Comment (sushi2k) [inline Document/0x04a-Android.md]: Two times "Content Providers", delete one.

File: Document/0x04a-Android.md [hunk 4]
```diff
@@ -160,6 +160,66 @@ A reference to the Messenger can be sent via an Intent using the previously ment
...
+*Activities*, *Services* and *Broadcast Receivers* can enforce a permission by entering the attribute *android:permission* inside each tag in AndroidManifest.xml:
+```
+<receiver
+    android:name="com.permissions.sample.AnalyticsReceiver"
+    android:enabled="true"
+    android:permission="com.permissions.sample.ACCESS_USER_INFO">
+    ...
+</receiver>
+```
+*Content Providers* are a little bit different it that they allow separate permissions for read, write or access the Content Provider using a content URI.
```
Comment (sushi2k) [inline Document/0x04a-Android.md]: Change "different it that" in "different in that"

**Comment by luander** [comment]
Thank you for all the comments. Will pay more attention to typos to avoid this inconvenience.

**Comment by sushi2k** [comment]
Can you still fix the <uses-permisison> to <uses-permission>, mentioned by @clviper?

**Comment by luander** [comment]
Fixed.

**Pull request merged**